klaviyo/klaviyo-swift-sdk
# PUSH-127 - Fix archiveRootObject deprecation

archiveRootObject is deprecated now. This removes use of that method. Also this introduces some testing infrastructure so the changes can be unit tested.

# Changes:

* Fix deprecation for archiving queue and unarchiving
* Add new methods for archive/unarchive and file utilities
* Add environment file to enable dependency injection
* Add tests for archive and file utilties
* Add test environment

# Testing
New unit tests added

## Manual Test Plan
Happy Path:
1. Run test app and force queue to be created.
2. Restart app and ensure queue is emptied

Upgrade Path
1. Run existing test app with current sdk and ensure queue is created.
2. Upgrade the sdk and run the app again ensure the queue is emptied.

# Release Plan
Update spm version and changelog
Update cocoapods version to 1.5.2 and publish it.
